File: app/src/main/java/com/chisw/switchmymind/network/NetworkApi.java
package com.chisw.switchmymind.network;

import com.chisw.switchmymind.network.model.InputData;
import com.chisw.switchmymind.network.model.Result;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;

/**
 * Created by user on 19.01.18.
 */

public interface NetworkApi {
    @Headers("Content-Type: application/json")
    @POST("/calculations")
    Call<Result> sendInputData(@Body InputData inputData);

    @Headers("Content-Type: application/json")
    @GET("/calculations")
    Call<Object> getResult();
}

File: app/src/main/java/com/unitalk/network/NetworkApi.kt
package com.unitalk.network

import com.unitalk.network.model.CardsClickedData
import com.unitalk.network.model.FirstSamplingResultsData
import com.unitalk.network.model.NextSamplingResultsData
import com.unitalk.network.model.UserData
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.Path

interface NetworkApi {
    @POST("/auth")
    fun sendAuthUserData(@Body userData: UserData): Call<Void>

    @POST("/calculations")
    fun sendFirstSampleUserData(@Body userData: UserData): Call<FirstSamplingResultsData>

    @POST("/calculations/{sessionId}")
    fun sendNextSampleUserData(@Path("sessionId") sessionId: String, @Body userData: UserData): Call<NextSamplingResultsData>

    @PATCH("/calculations/{calculationId}/pictures")
    fun sendUserCards(@Path("calculationId") calculationId: String, @Body cardsClickedData: CardsClickedData): Call<Void>
}